scripts/leapfrog/generate_resources.py

```diff
@@ -0,0 +1,322 @@
+    flavors = [
+        flavor for flavor in nova.flavors.list() if flavor.name == name
+    ]
+    if len(flavors) > 1:
+        raise Exception("Flavor name, '{}', is not unique.".format(name))
+    try:
+        return flavors[0].id
+    except IndexError:
+        raise Exception("No flavor with name '{}' exists.".format(name))
+
+
+def get_network_id_from_name(name):
+    neutron = neutron_client.Client(session=get_session())
+    networks_ = neutron.list_networks(name=name)
+    networks = networks_["networks"]
+    if len(networks) > 1:
+        raise Exception("Network name, '{}', is not unique.".format(name))
+    try:
+        return networks[0]["id"]
+    except IndexError:
+        raise Exception("No network with name '{}' exists.".format(name))
+
+
+def get_image_id_from_name(name):
+    glance = glance_client.Client(CLIENT_VERSION, session=get_session())
+    images = list(glance.images.list(name=name))
+    if len(images) > 1:
+        raise Exception("Image name, '{}', is not unique.".format(name))
+    try:
+        return images[0].id
+    except IndexError:
+        raise Exception("No image with name '{}' exists.".format(name))
+
+
+def verify_active(building, timeout=600):
+    nova = nova_client.Client(CLIENT_VERSION, session=get_session())
+    active = []
+    counter = 0
+    delay = 10
+    while building:
+        still_building = []
+        for server in building:
+            server = nova.servers.get(server.id)
+            if server.status == "ACTIVE":
+                active.append(server)
+            elif server.status == "BUILD":
+                still_building.append(server)
+            else:
+                raise Exception(
+                    "The resource {} is in an invalid state.".format(
+                        server.id
+                    )
+                )
+        building = still_building
+        counter += delay
+        if counter < timeout:
+            sleep(delay)
+        else:
+            raise Exception(
+                "All resources failed to build within {} seconds.".format(
+                    timeout,
+                )
+            )
+    return active
+
+
+def make_networks(conf):
+    neutron = neutron_client.Client(session=get_session())
+    net = IPv4Network(u'10.0.0.0/8')
+    snets = net.subnets(16)
+    ext_gateway = get_network_id_from_name(conf["external_gateway"])
+    for number, cidr in enumerate(snets):
+        network = neutron.create_network(
+            {
+                "network": {
+                    "name": "test_network_{}".format(number),
+                },
+            }
+        )
+        subnet = neutron.create_subnet(
+            {
+                "subnet": {
+                    "network_id": network["network"]["id"],
+                    "ip_version": 4,
+                    "cidr": cidr,
+                    "name": "test_subnet_{}".format(number),
+                },
+            }
+        )
+        router = neutron.create_router(
```

Comment:
Along the lines I mentioned above, if the pattern your suggesting is because of then some specific testing requirement I can change it. If not, what is the reason it would be better?